router/batchrouter/asyncdestinationmanager/sftp/utils.go

```diff
@@ -0,0 +1,313 @@
+		records = append(records, record)
+	}
+	return records, nil
+}
+
+func generateFile(filePath string, format string) (string, error) {
+	switch strings.ToLower(format) {
+	case "json":
+		return generateJSONFile(filePath)
+	case "csv":
+		return generateCSVFile(filePath)
+	default:
+		return "", errors.New("unsupported file format")
+	}
+}
+func generateJSONFile(filePath string) (string, error) {
+	// Parse JSON records
+	records, err := parseRecords(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	tmpFilePath, err := getTempFilePath()
+	if err != nil {
+		return "", err
+	}
+	tmpFilePath = fmt.Sprintf(`%v.json`, tmpFilePath)
+
+	// Create a temporary CSV file
+	tempFile, err := os.Create(tmpFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer tempFile.Close()
+
+	// Write JSON data to the temporary file
+	encoder := json.NewEncoder(tempFile)
+	encoder.SetIndent("", "  ")
+	err = encoder.Encode(records)
+	if err != nil {
+		return "", err
+	}
+
+	return tmpFilePath, nil
+}
+
+func generateCSVFile(filePath string) (string, error) {
+	// Parse JSON records
+	records, err := parseRecords(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	// Extract field names
+	fieldNames, err := getFieldNames(records)
+	if err != nil {
+		return "", err
+	}
+
+	tmpFilePath, err := getTempFilePath()
+	if err != nil {
+		return "", err
+	}
+	tmpFilePath = fmt.Sprintf(`%v.csv`, tmpFilePath)
+
+	// Create a temporary CSV file
+	tempFile, err := os.Create(tmpFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer tempFile.Close()
+
+	// Create a CSV writer
+	writer := csv.NewWriter(tempFile)
+
+	// Write header to the CSV file
+	if err := writer.Write(fieldNames); err != nil {
+		return "", err
+	}
+
+	// Write records to the CSV file
+	for _, record := range records {
+		var row []string
+		fields, ok := record["message"].(map[string]interface{})["fields"].(map[string]interface{})
+		if !ok {
+			return "", fmt.Errorf("fields not found in a record")
+		}
+		action, ok := record["message"].(map[string]interface{})["action"].(string)
+		if !ok {
+			return "", fmt.Errorf("action not found in a record")
+		}
+		fields["action"] = action
+		for _, key := range fieldNames {
+			row = append(row, fmt.Sprintf("%v", fields[key]))
+		}
+		if err := writer.Write(row); err != nil {
+			return "", err
+		}
+	}
+
+	// Flush any buffered data to the underlying writer
+	writer.Flush()
+	if err := writer.Error(); err != nil {
+		return "", err
+	}
+
+	return tmpFilePath, nil
+}
+
+func getTempFilePath() (string, error) {
+	localTmpDirName := fmt.Sprintf(`/%s/`, misc.RudderAsyncDestinationLogs)
+	tmpDirPath, err := misc.CreateTMPDIR()
+	if err != nil {
+		return "", err
+	}
+	folderPath := path.Join(tmpDirPath, localTmpDirName)
+	_, e := os.Stat(folderPath)
+	if os.IsNotExist(e) {
+		folderPath, _ = os.MkdirTemp(folderPath, "")
+	}
+	tmpFilePath := filepath.Join(folderPath, uuid.NewString())
+	return tmpFilePath, nil
+}
+
+func getUploadFilePath(path string) string {
```

Comment:
Do we have a unit test to cover this function? It seems like there are many cases inside and it could get complex.

Reply:
added the unit testcases